### PR TITLE
Extend DB class to provide a populate_metadata method

### PR DESCRIPTION
**Pull request by nerdstrike**
## Description

Extend DB class to provide a populate_metadata method which can xref_config.ini files. Additional relationships added to DBIC classes to support create_related use.

## Use case

Pipeline still uses a core script to load metadata, this can happen directly without shelling out

## Benefits

Removes dependencies, code is much tighter for consuming xref_config.ini

## Possible Drawbacks

Won't necessarily work with EG species not present in the test config file

## Testing

Simple unit tests added to ensure something was loaded into the relevant tables

_Have you run the entire test suite and no regression was detected?_

No interactions with other tests

### CODE REVIEW

**Comment by nerdstrike** [comment]
This isn't as clean as it could be, but it's holding back pipeline testing.

File: modules/Bio/EnsEMBL/Xref/DB.pm [hunk 1]
```diff
@@ -226,6 +234,255 @@ sub create_db_row {
...
+      }
+
+    }
+
+  }
+  return $config;
+}
+
+
+sub _mangle_source_block {
```
Comment (markmcdowall) [inline modules/Bio/EnsEMBL/Xref/DB.pm]: Documentation required

File: modules/Bio/EnsEMBL/Xref/DB.pm [hunk 2]
```diff
@@ -104,18 +111,19 @@ sub _init_db {
...
 
   my %deploy_opts = ();
   # Example deploy option $deploy_opts{add_drop_table} = 1;
-  print STDERR 'Connecting: '.$dsn."\n";
+  carp 'Connecting: '.$dsn."\n";
   my $schema = Bio::EnsEMBL::Xref::Schema->connect($dsn, $conf{user},$conf{pass}, \%opts);
 
   if ($conf{create} == 1 && $conf{driver} eq 'mysql') {
-    my $dbh = DBI->connect(sprintf("DBI:%s:database=;host=%s;port=%s",$conf{driver},$conf{host},$conf{port}),$conf{user},$conf{pass}, \%opts);
+    my $dbh = DBI->connect(sprintf('DBI:%s:database=;host=%s;port=%s',$conf{driver},$conf{host},$conf{port}),$conf{user},$conf{pass}, \%opts);
```
Comment (markmcdowall) [inline modules/Bio/EnsEMBL/Xref/DB.pm]: Could some spacing be added in these options just to improve the clarity, same for the `$dsn` statements above

**Comment by nerdstrike** [comment]
@markmcdowall - requested changes made

**Pull request merged**